### PR TITLE
Rename migrations to have a version number prefix

### PR DESCRIPTION
**Pull request by alex-page**
### WHY are these changes introduced?

We are creating a lot of migrations and it's not clear at a glance what version each migration is for.

### WHAT is this pull request doing?

 This PR updates all migrations to prefix them based on the version number.

### CODE REVIEW

**Review by sam-b-rose — APPROVED** [review]
Nice work! Love this version prefix convention

Only thing I would add here as well is updating the `plop` templates and prompt to include the prefix, either as an option or just as part of the input placeholder example.

Also to update this [section example](https://polaris.shopify.com/tools/polaris-migrator#creating-migrations) for creating a new migration. 👍

**Review by lgriffee — APPROVED** [review]

File: polaris-migrator/src/migrations/v9-styles-tokenize-space/v9-styles-tokenize-space.ts [hunk 1]
```diff
@@ -8,12 +8,12 @@ import {
...
   StopWalkingFunctionNodes,
-  createSassMigrator,
+  createSassMigrator as v9StylesTokenizeSpace,
 } from '../../utilities/sass';
 import {isKeyOf} from '../../utilities/type-guards';
 
-export default createSassMigrator(
-  'styles-tokenize-space',
+export default v9StylesTokenizeSpace(
+  'v10-styles-tokenize-space',
```
Comment (alex-page) [inline polaris-migrator/src/migrations/v9-styles-tokenize-space/v9-styles-tokenize-space.ts]:
```suggestion
  'v9-styles-tokenize-space',
```

File: polaris-migrator/src/migrations/v9-scss-replace-duration/v9-scss-replace-duration.ts [hunk 2]
```diff
@@ -22,7 +22,7 @@ const durationMap = {
   slowest: '--p-duration-500',
 };
 
-export default createSassMigrator(
+export default v9ScssReplaceDuration(
   'replace-sass-transition',
```
Comment (alex-page) [inline polaris-migrator/src/migrations/v9-scss-replace-duration/v9-scss-replace-duration.ts]:
```suggestion
export default v9ScssReplaceDuration(
  'v9-replace-sass-transition',
```

File: polaris-migrator/src/migrations/v9-scss-replace-easing/v9-scss-replace-easing.ts [hunk 3]
```diff
@@ -21,7 +21,7 @@ const easingMap = {
 
 const deprecatedEasingFuncs = ['anticipate', 'excite', 'overshoot'];
 
-export default createSassMigrator(
+export default v9ScssReplaceEasing(
   'scss-replace-easing',
```
Comment (alex-page) [inline polaris-migrator/src/migrations/v9-scss-replace-easing/v9-scss-replace-easing.ts]:
```suggestion
export default v9ScssReplaceEasing(
  'v9-scss-replace-easing',
```

**Review by aaronccasanova — APPROVED** [review]

File: .changeset/nine-news-guess.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+---
+'@shopify/polaris-migrator': minor
+'polaris.shopify.com': minor
```
Comment (alex-page) [inline .changeset/nine-news-guess.md]:
```suggestion
---
'@shopify/polaris-migrator': major
```

File: .changeset/nine-news-guess.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris-migrator': major
```
Comment (alex-page) [inline .changeset/nine-news-guess.md]:
```suggestion
---
'@shopify/polaris-migrator': minor
```

**Pull request merged**